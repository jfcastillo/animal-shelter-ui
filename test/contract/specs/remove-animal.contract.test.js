import { provider } from './init-pact';
import { AnimalController } from "../../../controllers";
import { Matchers } from "@pact-foundation/pact";

const animalName = "Lisa"
describe('Given an Animal Service', () => {
    // beforeAll(async () => {
    //     await provider.setup();
    // });
    describe('When a request to remove an animal is made', () => {
        beforeAll(async () => {
            await provider.setup();
            await provider.addInteraction({
                state: 'remove animal',
                uponReceiving: 'a request to remove an animal',
                withRequest: {
                    method: 'DELETE',
                    path: `/animals/${animalName}`,
                },
                willRespondWith: {
                    status: 204,

                }
            });
        });
        it("Then it should return the right data", async () => {
            const response = await AnimalController.delete(animalName);
            expect(response.data).toMatchSnapshot();
            await provider.verify();
        });
    });
    afterAll(async () => {
        await provider.finalize();
    })
});