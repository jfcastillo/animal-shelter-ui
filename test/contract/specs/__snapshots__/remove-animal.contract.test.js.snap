// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given an Animal Service When a request to remove an animal is made Then it should return the right data 1`] = `""`;
